### PR TITLE
Add blurb about dist tracing

### PR DESCRIPTION
**Pull request by JoshLove-msft**
- Addresses https://github.com/Azure/azure-sdk-for-net/issues/16238
- Update LineCounter sample app to show usage of Event Grid
- Update LineCounter README to include a screenshot of Azure Monitor
- Update Distributed Tracing section in Diagnostics readme to link to LineCounter.
- Also add in DI extension methods.

### CODE REVIEW

File: samples/linecounter/LineCounter.csproj [hunk 1]
```diff
@@ -13,4 +13,7 @@
   <ItemGroup>
     <Folder Include="Connected Services\" />
   </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\sdk\eventgrid\Azure.Messaging.EventGrid\src\Azure.Messaging.EventGrid.csproj" />
```
Comment (pakrym) [inline samples/linecounter/LineCounter.csproj:17]: I think we should use the package reference in samples when possible so they are independent of the repo.

File: samples/linecounter/Startup.cs [hunk 2]
```diff
@@ -27,8 +27,9 @@ public void ConfigureServices(IServiceCollection services)
 
                     builder.AddBlobServiceClient(Configuration.GetSection("Blob"));
                     builder.AddEventHubProducerClient(Configuration.GetSection("Uploads")).WithName("Uploads");
-
+
```
Comment (pakrym) [inline samples/linecounter/Startup.cs:30]: nit: ws

File: sdk/eventgrid/Azure.Messaging.EventGrid/api/Azure.Messaging.EventGrid.netstandard2.0.cs [hunk 3]
```diff
@@ -1341,3 +1341,12 @@ public partial class WebSlotSwapWithPreviewStartedEventData
         public string Verb { get { throw null; } }
     }
 }
+namespace Microsoft.Extensions.Azure
```
Comment (pakrym) [inline sdk/eventgrid/Azure.Messaging.EventGrid/api/Azure.Messaging.EventGrid.netstandard2.0.cs:1344]: "Add a blurb" you say. Sneaky :)

Do we want to try and add tests for these?

File: samples/linecounter/LineCounter.csproj (hunk 1 above)
Comment (JoshLove-msft) [reply samples/linecounter/LineCounter.csproj:17]: Yeah, I will update once we next release Event Grid. I can't do it yet as I am depending on the changes I have made to include the builder extensions.

Comment (JoshLove-msft) [reply samples/linecounter/LineCounter.csproj:17]: https://github.com/Azure/azure-sdk-for-net/issues/17138

File: sdk/eventgrid/Azure.Messaging.EventGrid/api/Azure.Messaging.EventGrid.netstandard2.0.cs (hunk 3 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/api/Azure.Messaging.EventGrid.netstandard2.0.cs:1344]: Created https://github.com/Azure/azure-sdk-for-net/issues/17158

**Review by pakrym — APPROVED** [review]

**Comment by baharedankoyuncu** [comment]
Hi @JoshLove-msft,

Out of curiosity, has these changes been published to NuGet?
I see the PR was merged but I haven't been able to use the newly added DI extension methods in startup and looking at the changelog I assume it hasn't been pushed out yet :-)

**Comment by JoshLove-msft** [comment]
> Hi @JoshLove-msft,
> 
> Out of curiosity, has these changes been published to NuGet?
> I see the PR was merged but I haven't been able to use the newly added DI extension methods in startup and looking at the changelog I assume it hasn't been pushed out yet :-)

Hi @baharedankoyuncu, 
This hasn't been published to NuGet yet. This will go out in our next release which would be in Feb.

**Comment by baharedankoyuncu** [comment]
Thank you for the quick reply @JoshLove-msft 👍

**Pull request merged**